src/compound_model.js

```diff
@@ -0,0 +1,165 @@
+import { InMemoryEntity } from "@exabyte-io/code.js/dist/entity";
+import { setNextLinks, setUnitsHead } from "@exabyte-io/code.js/dist/utils";
+
+import { UnitModel } from "./unit_model";
+import { UnitModelFactory } from "./unit_model_factory";
+import { buildModelPath } from "./utils";
+
+export class Model extends InMemoryEntity {
+    static UnitModelFactory = UnitModelFactory;
+
+    static UnitModelCls = UnitModel;
+
+    constructor({ ...config } = {}) {
+        super(config);
+        this._units = Model.instantiateUnits(config.units);
+        this.setNextLinks().setUnitsHead().syncToProp();
+    }
+
+    get name() {
+        return this.prop("name");
+    }
+
+    get slug() {
+        return this.prop("slug");
+    }
+
+    static isUnitModelArray(unitInstances) {
+        return (
+            Array.isArray(unitInstances) &&
+            unitInstances.every((u) => u instanceof Model.UnitModelCls)
+        );
+    }
+
+    static instantiateUnits(unitConfigs) {
+        if (!Array.isArray(unitConfigs) || !unitConfigs) return [];
+        return unitConfigs.map((config) => Model.UnitModelFactory.create(config));
+    }
+
+    /**
+     *  Get list of unit model _instances_
+     */
+    get units() {
+        return this._units;
+    }
+
+    /**
+     *  Set units and update `this._json`.
+     */
+    set units(unitInstances) {
+        const valid = Model.isUnitModelArray(unitInstances) || !unitInstances.length;
+        if (valid) {
+            this._units = unitInstances;
+            this.setNextLinks().setUnitsHead().syncToProp();
+        }
+    }
+
+    setUnitModels(unitInstances) {
+        this.units = unitInstances;
+        return this;
```

Comment:
We need a `ModelBuilder` if we want to follow this pattern. I suggest just explicitly calling all functions one by one in the constructor instead

Reply:
Also - this functionality setHead, setNextLinks is very similar between here and wode.js. We should consider implementing it in one place instead and reusing

Reply:
> We need a `ModelBuilder` if we want to follow this pattern. I suggest just explicitly calling all functions one by one in the constructor instead

A ModelBuilder could indeed be helpful. I used `return this` here mainly to allow for method chaining. I think this could be useful when we dynamically add a unit and want to filer/search/map etc right away.

Reply:
> Also - this functionality setHead, setNextLinks is very similar between here and wode.js. We should consider implementing it in one place instead and reusing

`setNextLinks` and `setUnitsHead` are already isolated to code.js